### ISSUE TITLE
Ignore-Missing-Schemas and Strict are incompatible

### ISSUE THREAD

**Issue by warmfusion** [instrumenta/kubeval#142]
Given a deployment that includes additional fields which are not to spec
When kubeval is executed with both `--strict` and `--ignore-missing-schemas` 
Then no errors are reported from the Deployment object

## Example output

```
$ kubeval build.yaml --strict --ignore-missing-schemas
Warning: Set to ignore missing schemas
Warning: Set to ignore missing schemas
Warning: Set to ignore missing schemas
The document build.yaml contains a valid Namespace
The document build.yaml contains a valid ConfigMap
The document build.yaml contains a valid Deployment
$ kubeval build.yaml --strict 
The document build.yaml contains a valid Namespace
The document build.yaml contains a valid ConfigMap
The document build.yaml contains an invalid Deployment
---> terminationGracePeriodSeconds: Additional property terminationGracePeriodSeconds is not allowed
```

I wonder this is a problem where if a schema is missing all fields are automatically 'skipped', however the Deployment object is a core structure, and should be a well-known object.

Note that we bundle our yaml into a single deliverable object and i'd like to check the bundle is valid but as we use Prometheus and Cert-Manager CRDs we need to ignore-missing-schemas otherwise the bundle check fails.

## Version
```
$ kubeval --version
Version: 0.11.0
Commit: 312975f44a29ebbce5d013c3b769d19e49aa1fb7
Date: 2019-07-13T17:40:49Z
```

**Comment by garethr** [comment]
This definitely sounds like a bug. Can you provide a sample `build.yaml` that triggers this? We'll get it fixed and a new release issued.

**Comment by warmfusion** [comment]
The following yaml is the smallest working reproduction I could manage.

if its not obvious the `invalid: argument` in spec.template.spec.containers[0] isn't valid 😄 

``` yaml
kind: Deployment
apiVersion: extensions/v1beta1
metadata:
  name: nginx-deployment
spec:
  template:
    spec:
      containers:
        - name: nginx
          invalid: argument
```

**Comment by ian-howell** [comment]
@warmfusion @garethr Fixed in #150

```
$ kubeval --strict --ignore-missing-schemas /tmp/deployment.yaml
Warning: Set to ignore missing schemas
The document /tmp/deployment.yaml contains an invalid Deployment
---> invalid: Additional property invalid is not allowed
```

**Comment by garethr** [comment]
Thanks to @ian-howell this is fixed. I'll get a new release out shortly with the fix.
